drivers/auxdisplay/auxdisplay_jhd1313.c

```diff
@@ -26,9 +26,9 @@ LOG_MODULE_REGISTER(auxdisplay_jhd1313, CONFIG_AUXDISPLAY_LOG_LEVEL);
 #define JHD1313_CS_RIGHT_SHIFT		(1 << 2)
 
 /* Defines for the JHD1313_CMD_INPUT_SET to change text direction */
-#define JHD1313_IS_SHIFT_INCREMENT	(1 << 1)
-#define JHD1313_IS_SHIFT_DECREMENT	(0 << 1)
-#define JHD1313_IS_ENTRY_LEFT		(1 << 0)
+#define JHD1313_IS_SHIFT_INCREMENT	(1 << 0)
```

Comment:
These came from https://github.com/zephyrproject-rtos/zephyr/blob/028e0a3c8d75fc5e2027f54dbb5a55aadbb1c490/include/zephyr/drivers/misc/grove_lcd/grove_lcd.h#L82 which shows increment as `1 << 1` not `1 << 0`, same for entry values. Since I copied and pasted and do not have this display, are you sure the new values are correct?

Reply:
Yes I've tested with an actual display. I took the values from https://github.com/Seeed-Studio/Grove_LCD_RGB_Backlight/blob/94758dfe5f68de1d40203b8922196e18a0ba9df3/rgb_lcd.h#L66-L69
Without this fix the text is backwards, i.e.
```
________________
___________OLLEH
```

BTW just realized that DECREMENT should probably be 0 << 0...

I agree this is confusing (and contradicts the datasheet, too) but all other drivers i could find on GitHub also have INCREMENT/DECREMENT on bit 0.

Reply:
Now that I think about it, could it be that all the other drivers are wrong? As they swapped the two bits, they ended setting ENTRY_LEFT|DECREMENT for roman languages... but it should probably be ENTRY_LEFT|INCREMENT, which if I'm not mistaken would have the same result as my "fix". Thoughts?

Reply:
For hd44780? https://lcd-linux.sourceforge.net/pdfdocs/hd44780.pdf
ID 0 = increment, ID 1 = decrement, S 1 = accompanies display shift, and the command has ID in bit 1 and S in bit 0, so the original bit positions to be valid

Reply:
> 	/* Initialize to the default text direction for romance languages */
>	cmd = JHD1313_IS_ENTRY_LEFT | JHD1313_IS_SHIFT_DECREMENT;

This doesn't make sense doing decrement by default, no, should be increment.

Reply:
Ya all the other drivers out there (all likely copy pasted from Seeed's original one, including this one) seem to have reproduced the same pattern, which in our case here is causing the bug since we have the right config flags where all the others had them reversed (also due to copy pasting, probably).

BTW this https://github.com/Seeed-Studio/Grove_LCD_RGB_Backlight/blob/master/rgb_lcd.cpp#L201-L211 looks wrong too and confirms the original inversion issue, as this should be using increment/decrement to reverse text flow, right?

Reply:
Agreed, that is wrong, surprised no-one has picked it up their since the function is named "leftToRight" and the flag being set is "ENTRYLEFT" not "ENTRYRIGHT"

Reply:
OK, just pushed what I think should be the right fix. RIGHT/LEFT has no meaning in the context of the Input Set. It's instead: move/don't move the cursor based on whether SHIFT is set or not. Whether it will grow to the left or to the right will depend on if it's INCREMENTing or DECREMENTing.